Enable CA2235 (non serializable fields on serializable types)

drewnoakes opened the pull request:
Relates to #4377

~These changes fix the errors, but looking at these I wonder whether the types wouldn't failed serialisation anyway, and if it wouldn't be better to just make them non-serialisable.~

davkean commented:
This is just noise as it just does it own serialization; implements GetObjectData.

Pilchie commented:
So, potentially use a local suppressmessage attribute on each of those types, instead of doing all the fields?

drewnoakes commented:
I believe the diagnostic to be incorrect, so opened:

- https://github.com/dotnet/roslyn-analyzers/issues/1970
- https://github.com/dotnet/roslyn/issues/32133 (related, also to #4415)

Will revisit this PR soon to get it cleaned up.

drewnoakes commented:
If https://github.com/dotnet/roslyn-analyzers/pull/1971 is merged then we can ignore suppression altogether.

drewnoakes commented:
It's merged, but the build's borked. 

https://devdiv.visualstudio.com/DevDiv/_build?definitionId=9244&_a=summary

drewnoakes commented:
The build recovered. I've updated the analyzer package to include dotnet/roslyn-analyzers#1971 which fixes the false positives.

The one remaining diagnostic was valid. Please review the type for which I removed `SerializableAttribute`.

jmarolf commented:
Looks good 👍
